baseapp/abci.go

```diff
@@ -24,6 +24,16 @@ import (
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
+const (
+	initialAppVersion = 0
+)
+
+var (
+	errAppVersionIsNotInitial = func(actualVersion uint64) error {
```

Comment:
I'm using the convention of storing errors and their messages as vars or constants respectively at the top of the file. It allows identifying these errors more easily and increases the likelihood of reusing them where needed.

The need for making this a function arises at times when we need to format parameters at runtime. That's why I create such anonymous functions for these "dynamic" errors. 

I'm not sure what the best practice that would allow the consolidation of all errors in one place while removing the need for such aliased functions with runtime parameters.

If you have a suggestion, please let me know

In terms of making this a regular function, I think it would make this error less visible and less likely to be reused.

Reply:
@peterbourgon 's suggestion [here](https://github.com/osmosis-labs/cosmos-sdk/pull/265#discussion_r899272899) seems like a better alternative for this

Reply:
> @peterbourgon 's suggestion [here](https://github.com/osmosis-labs/cosmos-sdk/pull/265#discussion_r899272899) seems like a better alternative than this aliased function

What he suggested is the idiomatic approach 👍